### PR TITLE
feat: add first set of allstars policies

### LINKED ISSUE #2: Add first set of Allstar policies
Creation of first iteration of allstar policies for otel-go & otel-go-contrib

### PR DESCRIPTION
**Pull request by sakshi-1505**
fixes #2

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
Thanks for opening this @sakshi-1505! really excited to see progress on this front! Just some questions as I'm not super familiar w/ the configuration presented.

**Comment by codeboten** [comment]
Pinging @open-telemetry/go-maintainers as well as this impacts their repos

File: binary_artifacts.yaml
Comment (sakshi-1505) [reply binary_artifacts.yaml:1]: They ensure that the project repo is free of any checked in binaries

```
This policy incorporates the [check from scorecard](https://github.com/ossf/scorecard/#scorecard-checks). Remove the binary artifact from the repository to achieve compliance. As the scorecard results can be verbose, you may need to run [scorecard itself](https://github.com/ossf/scorecard) to see all the detailed information.
```

File: admin.yaml
Comment (sakshi-1505) [reply admin.yaml:4]: This make sure that all the admin in repo are added via a Github-Team & there is no individual user-id added as Admin for any repo in an organisation.

File: actions.yaml
Comment (sakshi-1505) [reply actions.yaml:14]: Yes; & I have made this setting as is because I want the check to fail for a first time post which we can see what turn around time for Allstar is

File: binary_artifacts.yaml
Comment (codeboten) [reply binary_artifacts.yaml:1]: it would be great to have a brief comment at the top of each config file to make it clear what it is intended for 👍🏻

File: actions.yaml
Comment (sakshi-1505) [reply actions.yaml:1]: Done

**Review by jpkrohling — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Review by MrAlias — CHANGES_REQUESTED** [review]
What is allstars and why is it being applied to our repos?

It would be preferred if this change request was proposed after some kind of communication. Ideally synchronous. Can you attend next week's SIG meeting to describe what this is and why it is being applied?

**Comment by sakshi-1505** [comment]
> What is allstars and why is it being applied to our repos?
> 
> It would be preferred if this change request was proposed after some kind of communication. Ideally synchronous. Can you attend next week's SIG meeting to describe what this is and why it is being applied?

Hi Tyler; This is not exclusive to your repos but it is going to be rolled out to complete organisation. For the first pilot me & @jpkrohling had decided to go with otel-go repos. This is to mandatory implement an organisation wide governance policy which is a requirement from SIG-Security. You can read how & what Allstars will do on your repositories here https://github.com/open-telemetry/.allstar/blob/94009c49bc3e8483359187f74af891e477869614/README.md.
You can find more details around the requirement & why we went with Allstar here:
- https://github.com/open-telemetry/sig-security/issues/12
- https://github.com/open-telemetry/sig-security/issues/12#issuecomment-1699457358 https://github.com/open-telemetry/sig-security/issues/21

**Comment by jpkrohling** [comment]
@MrAlias, you are right, we should have pinged the Go folks before doing this. At this point, the SIG Security is testing this tool to check whether it can help us have visibility into the adherence of all repos in the organization to some (arbitrary) security practices. Once we know how this tool behaves, we'll be able to suggest it more broadly to all SIGs. We are not at that stage yet, but again, we should definitely have had your buy-in before using the Go repos for this trial.

In any case, I'll add a mention to this for the next maintainer's call and #otel-maintainers, so that you are all aware that we are experimenting with this.

**Comment by jpkrohling** [comment]
@MrAlias, I created a new thread on the #otel-go repo, so that other maintainers are aware: https://cloud-native.slack.com/archives/C01NPAXACKT/p1706539315137359

Is there any specific concern you'd like us to address?

**Comment by MrAlias** [comment]
> Is there any specific concern you'd like us to address?

What is allstar and what does it do? This PR description links to an issue that says it's going to set it up on our repository. There is no information about it and why this is being done.

I would like the impact this will have on our ability to uphold our [responsibilities](https://github.com/open-telemetry/community/blob/main/community-membership.md#responsibilities-and-privileges-3) to be explained. When I joined this project maintainer are the ones responsible for the success and health of their repositories. This PR is being added by a security SIG that does not share that responsibility nor have a charter to do so.

I would also like to understand how this decision was made in such a vacuum. I expect this is not going to be the last of such decisions and it really deserves a retrospective. Having an external SIG impose decisions on existing SIGs with independent authority is not the way to run this project. Both the security SIG and governance committee need to address this.

**Comment by pellared** [comment]
From https://github.com/ossf/allstar:

> For some security policies, Allstar can also automatically change the project setting that caused the violation, reverting it to the expected state.

I think we should make sure that it is NOT able to do any changes to the project settings. Otherwise, we are increasing the attack surface. This tool have permissions to be only readonly.

> The app asks for [...] file contents 

This is concerning me a lot 😬

File: allstar.yaml
Comment (MadVikingGod) [inline allstar.yaml:8]: Should we also include `opentelemetry-go-build-tools` and `opentelemetry-proto-go`?

**Comment by MadVikingGod** [comment]
This was discussed at the SIG today:
- We would like some more info on how this decision was made.  With just surface exploration, we have seen extensive permissions that seem to be a risk.  So, how this was evaluated vs other solutions would be helpful in making this decision.
- We would like some way to be notified of changes to our repositories, including if there is new scope or if this changes from `issues` to `fix`.
- Is there some process that you have for going from creating issues to automatic enforcement?  We sometimes have trouble doing tasks that require admin permissions. We don't want to be fighting a bot that removes that.
- Is it possible to rollback the automatic enforcement in case something goes wrong?
- Could we start this with a setting as `log`?  That would mean that the Security Sig would need to review it instead of pushing this onto the maintainer immediately.

**Comment by jpkrohling** [comment]
> We would like some more info on how this decision was made

That's fair, and given it's not the first time this was asked, I decided to follow the paper trail to give a more complete picture here: the Security SIG was [established last year](https://github.com/open-telemetry/community/pull/1576) with the [goal of assisting maintainers](https://github.com/open-telemetry/sig-security/blob/main/sig-security-charter.md) in the security-related practices, and one of the things we are working on is a set of _recommendations_ for maintainers, given that not everyone is expected to be knowledgeable in the security domain. 

According to our notes for the [meeting from August 23](https://docs.google.com/document/d/1P2xejC7lEkOV_Z-8E0oZPXLK5HOnUPNuRqKP0ZQ5fpg/edit#heading=h.dyjf5kjukp2h), @codeboten brought this to the issue to the maintainers' call [earlier that week](https://docs.google.com/document/d/14lBtLAdzsV7M8xQNdhTJ0d2PxM1R7FByw-GttKJzVVw/edit#heading=h.wtnirgc39er6) and it was well received. On that call, he shared the following tracker issue: https://github.com/open-telemetry/sig-security/issues/12 . This is the one pertaining Go: https://github.com/open-telemetry/opentelemetry-go/issues/4459

One week later, while talking about ways to do this on a bigger scale, we were pointed out to Allstar, which we started to investigate (https://github.com/open-telemetry/sig-security/issues/21).

> With just surface exploration, we have seen extensive permissions that seem to be a risk

I understand we've given read-only permissions to this app. I'll let @sakshi-1505 confirm if that's all we need, and I'll let someone from the TC double-check. Our TC sponsor is @reyang, with @arminru working closely with us on this.

> We would like some way to be notified of changes to our repositories

The SIG Security has no permission nor intention to change anything in anyone's repositories. All changes go through the TC, like this: https://github.com/open-telemetry/community/issues/1831 . 

In retrospect, I should have touched base with the Go SIG maintainers before including their repos during the installation of the app in the organization and I should have asked maintainers first regarding who'd be on board for trialing this tool.

> Is there some process that you have for going from creating issues to automatic enforcement?

We don't plan on having automatic enforcement. The tool will create an issue if it sees a deviation from our recommendations. That's exactly what the humans of the SIG Security would also do manually. If a specific issue is seen as a false-positive, we can remove the specific repo from that specific check.

> Is it possible to rollback the automatic enforcement in case something goes wrong?

No automatic enforcement is planned.

For the moment, we'll remove the Go repositories from the application installed at the organization level, adding only the Java Auto instrumentation, which volunteered for this trial.

**Comment by sakshi-1505** [comment]
> I understand we've given read-only permissions to this app. I'll let @sakshi-1505 confirm if that's all we need, and I'll let someone from the TC double-check. Our TC sponsor is @reyang, with @arminru working closely with us on this.

We don't need the intrusive permissions; those are only required when you want Allstar to have autofix for the issues but we are not doing that since we want the issues to get logged & manual intervention on fixes.

>Could we start this with a setting as log? That would mean that the Security Sig would need to review it instead of pushing this onto the maintainer immediately.

This setting is a no-op from a user point of view until & unless one self-host their Allstar application. The log in Allstar policy refers to the application log from where one can set custom alerts to get notified.

**Comment by sakshi-1505** [comment]
@jpkrohling @codeboten Can you folks please re-check & merge accordingly?

**Review by codeboten — APPROVED** [review]
Thanks @sakshi-1505! @MrAlias I believe your request for change has been addressed correct?

**Comment by MrAlias** [comment]
> @MrAlias I believe your request for change has been addressed correct?

Yeah. I don't see a way for me to dismiss my review. Please dismiss my review if you're able.

**Pull request merged**
